odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,538 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Operators of linearized deformations.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator.operator import Operator
+import odl
+import numpy as np
+
+
+__all__ = ('LinDeformFixedTempl', 'LinDeformFixedTemplDeriv',
+           'LinDeformFixedTemplDerivAdj', 'LinDeformFixedDisp',
+           'LinDeformFixedDispAdj', 'linear_deform', 'deform_grad')
```

Comment:
There is a difference between "public" and "reachable". Things you put into `__all__` are **public** and **prominent**, which means in practice that they are "imported up" to the highest level available, here probably to `odl.deform`. That means in particular that users will get them in the auto-completion list when they start typing `odl.deform <TAB>`. Hence, they appear very prominently and clearly convey the impression that they are intended for use by everybody.
On the other hand, functions that are not in `__all__` but do not start with an underscore are still to be considered **public**, but maybe not as important.
Finally, there are the things that have a name starting with an underscore. These are to be considered **private** or **internal**, although there is no way to *enforce* this privateness. 
In general, however, it is safe to say that Python is very bad at denying access to attributes, functions or classes since there is nothing like strictly "private" or "protected". If you want to access something in a module or class, there is a way to achieve this.
To make a long story short, even if you don't put stuff into `__all__`, it is still **reachable**. If you type `from odl.deform.linearized_deform import linear_deform` you will get the function, no matter if it's in `__all__` or not. Or, if you `import odl`, you will be able to access the above function via `odl.deform.linearized_deform.linear_deform`. One of these two possibilities is what you want to use in tests.